### PR TITLE
10b. how to use the kv store from nim and SDKs

### PR DESCRIPTION
**Pull request by JamieDawson**
This documents how to use the kv from the cli and from the sdks.

10b

### CODE REVIEW

File: docs/key-value-storage.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+Your Nimbella account has been provisioned with Key-Value Storage and Object Storage. In this document, we are going to cover Key-Value Storage.
+
+A Key-Value Storage acts as a database where you create keys and apply values to those keys.
```
Comment (rabbah) [inline docs/key-value-storage.md:3]:
```suggestion
A Key-Value Store, or KV for short, acts as a database where you create keys and apply values to those keys.
```

File: docs/key-value-storage.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+The `nim` command will show your current content.
+
+```
+# To show the list of key-values (may be empty initially)
```
Comment (rabbah) [inline docs/key-value-storage.md:8]:
```suggestion
```

File: docs/key-value-storage.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+Your Nimbella account has been provisioned with Key-Value Storage and Object Storage. In this document, we are going to cover Key-Value Storage.
+
+A Key-Value Storage acts as a database where you create keys and apply values to those keys.
+
+The `nim` command will show your current content.
```
Comment (rabbah) [inline docs/key-value-storage.md:5]:
```suggestion
The following `nim` command will show the current contents of the KV. The command will not show any output if the KV is empty.
```

File: docs/key-value-storage.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+In this visual example, you can store user info, such as their City, State, and Country, and apply the correct values to them. This allows you to keep track of data and request the data when needed.
+
+![](https://live.staticflickr.com/65535/50653417258_e27306c044_k.jpg)
+
+### Using a Key-Value Storage inside actions using a Nimbella SDK
+
+Nimbella provides SDKs for use within actions. Currently, these exist for `nodejs`, `php`, and `python`. An SDK is under development for `java` and the plan is to eventually cover all supported languages.
+
+### Using the `nodejs` SDK
```
Comment (rabbah) [inline docs/key-value-storage.md:33]: You'll want to convert these into language tabbed views. See this for an example https://github.com/nimbella/docs/pull/13.

File: docs/key-value-storage.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+- Conceptual Overview
+- Using a Key-Value Storage inside actions using a Nimbella SDK
+  - The `nodejs` SDK
+  - The `php` SDK
+  - The `python` SDK
+- Common Redis commands
+
+### Conceptual Overview
+
+A Key-Value Storage behaves like a database where you can create keys, apply values to those keys, and then call on them to extract the data that you need.
```
Comment (rabbah) [inline docs/key-value-storage.md:23]: replace Key-Value Storage with KV or Key-Value Store when uses as a noun.

File: docs/key-value-storage.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+In this visual example, you can store user info, such as their City, State, and Country, and apply the correct values to them. This allows you to keep track of data and request the data when needed.
+
+![](https://live.staticflickr.com/65535/50653417258_e27306c044_k.jpg)
+
+### Using a Key-Value Storage inside actions using a Nimbella SDK
+
+Nimbella provides SDKs for use within actions. Currently, these exist for `nodejs`, `php`, and `python`. An SDK is under development for `java` and the plan is to eventually cover all supported languages.
```
Comment (rabbah) [inline docs/key-value-storage.md:31]: Please link to each SDK repo.

File: docs/key-value-storage.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+### Conceptual Overview
+
+A Key-Value Storage behaves like a database where you can create keys, apply values to those keys, and then call on them to extract the data that you need.
+
+In this visual example, you can store user info, such as their City, State, and Country, and apply the correct values to them. This allows you to keep track of data and request the data when needed.
+
+![](https://live.staticflickr.com/65535/50653417258_e27306c044_k.jpg)
+
+### Using a Key-Value Storage inside actions using a Nimbella SDK
+
```
Comment (rabbah) [inline docs/key-value-storage.md:30]: We need a note here such as "Nimbella KV uses Redis as the underlying implantation. Redis is ...".
This is to transition the reader to understand why we talk about KV then they'll see "redis" next.

File: docs/key-value-storage.md (hunk 4 above)
Comment (JamieDawson) [reply docs/key-value-storage.md:53]: I tried to follow along with that. I pushed my changes and I'm hoping I did everything correctly.

**Comment by JamieDawson** [comment]
All suggestions have been applied.

**Comment by JamieDawson** [comment]
Updated repo to have an image in the folder instead of Flickr.

**Comment by rabbah** [comment]
@JamieDawson I tried to build this locally (using `yarn start`) and got a syntax error:
```
./docs/key-value-storage.md
SyntaxError: docs/key-value-storage.md: Unexpected token (16:0)

  14 | {label: 'PHP', value: 'php'},
  15 | ]}
> 16 | <blockquote>
     | ^
  17 | </blockquote>
  18 | <p>{`Your Nimbella account has been provisioned with Key-Value Storage and Object Storage. In this document, we are going to cover Key-Value Storage.`}</p>
  19 | <p>{`A Key-Value Store, or KV for short, acts as a database where you create keys and apply values to those keys.`}</p>
```

**Comment by JamieDawson** [comment]
1) Broken code fixed.
2) Images replaced with local images.
3) Code snippets labeled with languages. 
4) Connected doc to the sidebar.

Pending review.

**Review by rabbah — APPROVED** [review]

**Pull request merged**